features/admin.connections.v1/pages/authenticator-edit.tsx

```diff
@@ -0,0 +1,371 @@
+import React, { Fragment, FunctionComponent, ReactElement, ReactNode, useEffect, useRef, useState } from "react";
+import { useTranslation } from "react-i18next";
+import { useDispatch, useSelector } from "react-redux";
+import { RouteComponentProps } from "react-router";
+import { Dispatch } from "redux";
+import { getLocalAuthenticator } from "../api/authenticators";
+import { useGetConnectionTemplate } from "../api/connections";
+import { EditConnection } from "../components/edit/connection-edit";
+import { CommonAuthenticatorConstants } from "../constants/common-authenticator-constants";
+import { AuthenticatorMeta } from "../meta/authenticator-meta";
+import { AuthenticatorLabels } from "../models/authenticators";
+import { CustomAuthConnectionInterface } from "../models/connection";
+
+/**
+ * Proptypes for the Custom Local Authenticator edit page component.
+ */
+type AuthenticatorEditPagePropsInterface = IdentifiableComponentInterface & RouteComponentProps;
+
+/**
+ * Custom Local Authenticator Edit page.
+ *
+ * @param props - Props injected to the component.
+ *
+ * @returns React element.
+ */
+export const AuthenticatorEditPage: FunctionComponent<AuthenticatorEditPagePropsInterface> = ({
+    location,
+    "data-componentid": _componentId = "authenticator-edit-page"
+}: AuthenticatorEditPagePropsInterface): ReactElement => {
+    const dispatch: Dispatch = useDispatch();
+
+    const idpDescElement: React.MutableRefObject<HTMLDivElement> = useRef<HTMLDivElement>(null);
+
+    const { CONNECTION_TEMPLATE_IDS: ConnectionTemplateIds } = CommonAuthenticatorConstants;
+
+    const [ connector, setConnector ] = useState<CustomAuthConnectionInterface>(undefined);
+    const [ isConnectorDetailsFetchRequestLoading, setConnectorDetailFetchRequestLoading ] = useState<boolean>(undefined);
+    const [ tabIdentifier, setTabIdentifier ] = useState<string>();
+    const [ isAutomaticTabRedirectionEnabled, setIsAutomaticTabRedirectionEnabled ] = useState<boolean>(false);
+    const [ shouldFetchConnectionTemplate, setShouldFetchConnectionTemplate ] = useState<boolean>(false);
+    const [ templateId, setTemplateId ] = useState<string>(undefined);
+
+    const { getLink } = useDocumentation();
+    const { t } = useTranslation();
+
+    const applicationsFeatureConfig: FeatureAccessConfigInterface = useSelector(
+        (state: AppState) => state?.config?.ui?.features?.applications
+    );
+    const featureConfig: FeatureConfigInterface = useSelector((state: AppState) => state.config.ui.features);
+    const productName: string = useSelector((state: AppState) => state?.config?.ui?.productName);
+
+    const isReadOnly: boolean = !useRequiredScopes(featureConfig?.identityProviders?.scopes?.update);
+    const hasApplicationTemplateViewPermissions: boolean = useRequiredScopes(applicationsFeatureConfig?.scopes?.read);
+
+    const { data: template } = useGetConnectionTemplate(templateId, shouldFetchConnectionTemplate);
```

Comment:
We can directly pass `!!templateId` instead of defining a new state variable and an effect function.

```suggestion
    const { data: template } = useGetConnectionTemplate(templateId, !!templateId);
```